CMakeLists.txt

```diff
@@ -118,6 +119,17 @@ set( TRACCC_THRUST_OPTIONS "" CACHE STRING
 mark_as_advanced( TRACCC_THRUST_OPTIONS )
 thrust_create_target( traccc::Thrust ${TRACCC_THRUST_OPTIONS} )
 
+option( TRACCC_USE_SYSTEM_ROCTHRUST
+   "Pick up an existing installation of rocThrust from the build environment"
+   ${TRACCC_USE_SYSTEM_LIBS} )
+if( TRACCC_BUILD_HIP )
```

Comment:
You will need a separate cache variable here, not `TRACCC_BUILD_HIP`.

The logic with all of our externals is that we use 2 cache variables with all of them.
  - `TRACCC_SETUP_<FOO>`: Flag controlling whether traccc should do **anything** to set up the external in question;
  - `TRACCC_USE_SYSTEM_<FOO>`: Flag controlling whether to set up `<FOO>` through `find_package(...)` or by building it ourselves.

The first one allows us to build traccc as part of an even higher-level project. Either Acts, or Athena. (Both of which we exercise already.) And allow Acts / Athena to set up a rocThrust version that they want.

Long story short, please introduce `TRACCC_SETUP_ROCTHRUST`, and use that here. `TRACCC_BUILD_HIP` should just be used to decide whether the HIP code inside of traccc would be built. Not what externals would be set up.